models/yang/sonic/sonic-sflow.yang

```diff
@@ -86,8 +96,10 @@ module sonic-sflow {
                     }
 
                     leaf agent_id {
-                       sonic-ext:custom-validation ValidateSflowAgentId;
-                       type string;
+                        type leafref {
+                            path "/prt:sonic-port/prt:PORT/prt:PORT_LIST/prt:ifname";
+                        }
+                        sonic-ext:custom-validation ValidateSflowAgentId;
```

Comment:
Is this custom validation needed ? Could not find the code.

Reply:
You can find the code in this commit. https://github.com/project-arlo/sonic-mgmt-framework/commit/890d4c78797a9052aee49f780bfa8f9a809e1ad4